satpy/scene.py

```diff
@@ -341,56 +346,69 @@ def all_dataset_ids(self, reader_name=None, composites=False):
-        dep_tree = DependencyTree(self.readers, comps, mods)
-        dep_tree.find_dependencies(set(available_datasets + all_comps))
+        # wishlist from self.dep_tree
+        dep_tree = DependencyTree(self.readers, sensor_comps, mods, available_only=True)
+        # ignore inline compositor dependencies starting with '_'
+        comps = (comp for comp_dict in sensor_comps.values()
+                 for comp in comp_dict.keys() if not comp.name.startswith('_'))
+        # make sure that these composites are even create-able by these readers
+        all_comps = set(comps)
+        # find_dependencies will update the all_comps set with DatasetIDs
+        dep_tree.find_dependencies(all_comps)
         available_comps = set(x.name for x in dep_tree.trunk())
         # get rid of modified composites that are in the trunk
         return sorted(available_comps & set(all_comps))
 
-    def available_composite_names(self, available_datasets=None):
+    def available_composite_ids(self):
+        """Get names of compositors that can be generated from the available datasets.
+
+        Returns: list of available compositor's names
+        """
+        return self._check_known_composites(available_only=True)
+
+    def available_composite_names(self):
         """All configured composites known to this Scene."""
-        return sorted(set(x.name for x in self.available_composite_ids(
-            available_datasets=available_datasets)))
+        return sorted(set(x.name for x in self.available_composite_ids()))
 
-    def all_composite_ids(self, sensor_names=None):
-        """Get all composite IDs that are configured.
+    def all_composite_ids(self):
+        """Get all IDs for configured composites.
 
-        Returns: generator of configured composite names
+        Returns: list of configured composite names
         """
-        if sensor_names is None:
-            sensor_names = self.attrs['sensor']
-        compositors = []
-        # Note if we get compositors from the dep tree then it will include
-        # modified composites which we don't want
-        for sensor_name in sensor_names:
-            sensor_comps = self.cpl.compositors.get(sensor_name, {}).keys()
-            # ignore inline compositor dependencies starting with '_'
-            compositors.extend(c for c in sensor_comps if not c.name.startswith('_'))
-        return sorted(set(compositors))
+        return self._check_known_composites()
+
+    def all_composite_names(self):
+        """Get all names for all configured composites.
 
-    def all_composite_names(self, sensor_names=None):
-        return sorted(set(x.name for x in self.all_composite_ids(sensor_names=sensor_names)))
+        Args:
+            reader_name (str): Limit composites possible to only those created
+                with data from the specified reader.
+            sensor_names (list): Limit composites to those configured for the
+                specified sensors.
```

Comment:
The method doesn't take any arguments.

Reply:
Thank you this was left over from an earlier iteration of changes.